src/js/media-trimmer.js

```diff
@@ -0,0 +1,493 @@
+      this.media.currentTime = timestampForClick;
+    }
+  }
+
+  mediaSetCallback(media) {
+    super.mediaSetCallback(media);
+
+    this._timeUpdated = this.timeUpdated.bind(this);
+    media.addEventListener('timeupdate', this._timeUpdated);
+
+    /*
+    // If readyState is supported, and the range is used before
+    // the media is ready, use the play promise to set the time.
+    if (media.readyState !== undefined && media.readyState == 0) {
+      // range.addEventListener('change', this.playIfNotReady);
+    }
+
+    // TODO: Update value if video already played
+    media.addEventListener('progress', this.updateBar.bind(this));
+    */
+
+    // Initialize thumbnails
+    if (media.textTracks && media.textTracks.length) {
+      const thumbnailTrack = Array.prototype.find.call(media.textTracks, t => t.label == 'thumbnails');
+
+      if (thumbnailTrack) {
+        this.enableThumbnails();
+      } else {
+        this.thumbnailPreview.style.display = 'none';
+      }
+    }
+  }
+
+  timeUpdated (evt) {
+    const percentComplete = lockBetweenZeroAndOne(this.media.currentTime / this.media.duration);
+    const fullW = this.wrapper.getBoundingClientRect().width;
+    const progressW = (percentComplete * fullW);
+
+    this.playhead.style.left = `${progressW}px`;
+    this.playhead.style.display = 'block';
+
+    /*
+     * if paused, we don't need to do anything else
+     */
+    if (!this.media.paused) {
+      const { startTime, endTime } = this.getCurrentClipBounds();
+
+      if (this.media.currentTime < startTime) {
+        this.media.currentTime = startTime;
+      }
+      if (this.media.currentTime > endTime) {
+        this.media.currentTime = startTime;
+      }
+      console.log('debug timeUpdated', this.media.currentTime, startTime, endTime);
+    }
+  }
+
+  mediaUnsetCallback(media) {
+    super.mediaUnsetCallback(media);
+
+    media.removeEventListener('timeupdate', this._timeUpdated);
+    this.wrapper.removeEventListener('click', this._clickHandler);
+
+    this.wrapper.removeEventListener('touchstart', this._dragStart);
+    this.wrapper.removeEventListener('touchend', this._dragEnd);
+    this.wrapper.removeEventListener('touchmove', this._drag);
+
+    this.wrapper.removeEventListener('mousedown', this._dragStart);
+    this.wrapper.removeEventListener('mouseup', this._dragEnd);
+    this.wrapper.removeEventListener('mousemove', this._drag);
+
+  }
+
+  /* Add a buffered progress bar */
+  getBarColors() {
+    const media = this.media;
+    let colorsArray = super.getBarColors();
+
+    if (!media || !media.buffered || !media.buffered.length || media.duration <= 0) {
+      return colorsArray;
+    }
+
+    const buffered = media.buffered;
+    const buffPercent = (buffered.end(buffered.length-1) / media.duration) * 100;
+    colorsArray.splice(1, 0, ['var(--media-progress-buffered-color, #777)', buffPercent]);
+    return colorsArray;
+  }
+
+  enableThumbnails() {
```

Comment:
In the future we might be able to break this out into a shared add on.

Reply:
agreed